core/src/main/java/org/elasticsearch/index/query/FuzzyQueryBuilder.java

```diff
@@ -19,35 +19,54 @@
 
 package org.elasticsearch.index.query;
 
+import org.apache.lucene.index.Term;
+import org.apache.lucene.search.FuzzyQuery;
+import org.apache.lucene.search.MultiTermQuery;
+import org.apache.lucene.search.Query;
+import org.elasticsearch.common.Nullable;
+import org.elasticsearch.common.Strings;
+import org.elasticsearch.common.io.stream.StreamInput;
+import org.elasticsearch.common.io.stream.StreamOutput;
 import org.elasticsearch.common.unit.Fuzziness;
 import org.elasticsearch.common.xcontent.XContentBuilder;
+import org.elasticsearch.index.mapper.MappedFieldType;
+import org.elasticsearch.index.query.support.QueryParsers;
 
 import java.io.IOException;
+import java.util.Objects;
 
 /**
  * A Query that does fuzzy matching for a specific value.
  */
-public class FuzzyQueryBuilder extends MultiTermQueryBuilder implements BoostableQueryBuilder<FuzzyQueryBuilder> {
+public class FuzzyQueryBuilder extends MultiTermQueryBuilder<FuzzyQueryBuilder> implements BoostableQueryBuilder<FuzzyQueryBuilder> {
 
     public static final String NAME = "fuzzy";
 
-    private final String name;
+    public static final Fuzziness DEFAULT_FUZZINESS = Fuzziness.AUTO;
 
-    private final Object value;
+    public final static int DEFAULT_PREFIX_LENGTH = FuzzyQuery.defaultPrefixLength;
 
-    private float boost = -1;
+    public final static int DEFAULT_MAX_EXPANSIONS = FuzzyQuery.defaultMaxExpansions;
 
-    private Fuzziness fuzziness;
+    public final static boolean DEFAULT_TRANSPOSITIONS = false;
 
-    private Integer prefixLength;
+    private final String fieldName;
 
-    private Integer maxExpansions;
+    private final Object text;
```

Comment:
If this stays an Object we have a problem with builder equality when we do the roundtrip doXContent->fromXContent that we currently do in the tests. I had similar problems in TermQueryBuilder already. Imagine an Integer value her (like its used in the current test setup), that is rendered to JSON (or other XContent) via `writeValue(Object value)` which inspects the class of the Object and then chooses a type-specific method to write it. On the parser side, this is always read back as a String with `parser.text()`. So if we check those builders for equality we will get `126 != "126"`. For the lucene query produced in the end this won't matter, but I think it would be great to keep that property that when we have builder A, generate XContent for that and parse it to Builder B, they should be the same.
One option here would be to push up the conversion to String we do in `toQuery()` anyway to the constructor already. Also I'm not sure if the java API currently is to lenient here by allowing any `Object`, since only objects that correctly parse back as Strings will really work with the current parser code. Another option might be to allow parsing to numeric values using `objectText()` like in CommonTermsQueryParser.